recipes/catch2/2.x.x/test_package/conanfile.py

```diff
@@ -10,6 +10,7 @@ class TestPackageConan(ConanFile):
     def build(self):
         cmake = CMake(self)
         cmake.definitions["WITH_MAIN"] = self.options["catch2"].with_main
+        cmake.definitions["WITH_BENCHMARK"] = self.options["catch2"].with_main and self.options["catch2"].with_benchmark
```

Comment:
Can `with_benchmark=True` and `with_main=False`?
If not, you'll need to add a test to `validate` of the main recipe..
Also, you can then simply do `cmake.definitions["WITH_BENCHMARK"] = self.options["catch2"].with_benchmark` here.

Reply:
if `with_main=True` then the library is built as a static library. So in such cases when benchmarking code is required it needs to be done when the library is built. if `with_main=False` then its a header only library and the user can choose to enable the benchmarking code themselves. Thats why the current code is conditional on both being True. I could add the test case for benchmark when `with_main=False`.

Reply:
I have added the test cases for benchmark  when `with_main=False.`

Reply:
> you'll need to add a test to `validate` of the main recipe

This suggestion seems to be required